### PR TITLE
Fix minor doc typos

### PR DESCRIPTION
**Pull request by gil2rok**
**Overview:** Two minor documentation changes on the CmdStanPy workflow documentation page

- remove period from middle of sentence
- make link formatting consistent

for the optimization inference algorithm.

**Details:** More specifically, the documentation for every Stan inference algorithm on this documentation page, follows the form of "Stan's *some algorithm*", with a link to the algorithm itself.

However, the optimization algorithm is the only type of inference algorithm that includes the word "Stan" as part of the link. No other inference algorithm does this.

#### Submission Checklist

- [ ] Run unit tests
- [x] Declare copyright holder and open-source license: see below

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

File: docsrc/users-guide/workflow.rst [hunk 1]
```diff
@@ -145,8 +145,8 @@ An example of each is provided in the `next section <examples.rst>`__.
 
   It returns a :class:`CmdStanVB` object which contains an approximation the posterior distribution.
 
-* The :meth:`~CmdStanModel.optimize` runs one of
-  `Stan's optimization algorithms <https://mc-stan.org/docs/reference-manual/optimization-algorithms-chapter.html>`__.
+* The :meth:`~CmdStanModel.optimize` runs one of Stan's
+  `optimization algorithms <https://mc-stan.org/docs/reference-manual/optimization-algorithms-chapter.html>`__
```
Comment (bob-carpenter) [inline docsrc/users-guide/workflow.rst:149]: I'm not sure why you're escaping the name and link as code here.  Or in the original.  I'd just say 

```
runs one of Stan's *optimization algorithms* <https://mc-stan.org/docs/reference-manual/optimization-algorithms-chapter.html>.  
```

which renders as

> runs one of Stan's *optimization algorithms* <https://mc-stan.org/docs/reference-manual/optimization-algorithms-chapter.html>.

Or if you can use markdown for links, I'd just use

```
runs one of Stan's [*optimization algorithms*](https://mc-stan.org/docs/reference-manual/optimization-algorithms-chapter.html)
```

which renders as

> runs one of Stan's [*optimization algorithms*](https://mc-stan.org/docs/reference-manual/optimization-algorithms-chapter.html)

**Review by bob-carpenter — CHANGES_REQUESTED** [review]
question about why not just linking directly and why escaping non-code as code

File: docsrc/users-guide/workflow.rst (hunk 1 above)
Comment (WardBrian) [reply docsrc/users-guide/workflow.rst:149]: This is how you structure a link in ReStructuredText

Comment (bob-carpenter) [reply docsrc/users-guide/workflow.rst:149]: But why the extra space before the periods?  Is that required?  This doc has two and the other PR from @WardBrian uses one.  I'd rather see none if that's legal!

**Pull request merged**